tests/difference/core/quint_model/ccv.qnt

```diff
@@ -0,0 +1,877 @@
+        pure val newChainState = currentState.providerState.chainState.with(
+            "currentValidatorSet", newValidatorSet
+        )
+        currentState.with(
+            "providerState", 
+                currentState.providerState.with(
+                    "chainState", newChainState
+            )
+        )
+    }
+
+    // Returns true if the given chain is currently a running consumer, false otherwise.
+    pure def isCurrentlyConsumer(chain: Chain, providerState: ProviderState): bool = {
+        val status = providerState.consumerStatus.get(chain)
+        status == RUNNING
+    }
+
+    // Returns the set of all consumer chains that currently have the status RUNNING.
+    pure def getRunningConsumers(providerState: ProviderState): Set[Chain] = {
+        providerState.consumerStatus.keys().filter(
+            chain => providerState.consumerStatus.get(chain) == RUNNING
+        )
+    }
+
+    // Returns the set of all consumer chains that currently have the status UNUSED.
+    pure def getUnusedConsumers(providerState: ProviderState): Set[Chain] = {
+        providerState.consumerStatus.keys().filter(
+            chain => providerState.consumerStatus.get(chain) == UNUSED
+        )
+    }
+
+    // Returns whether the consumer has timed out due to the VSCTimeout, and an error message.
+    // If the second return is not equal to "", the first return should be ignored.
+    // If it is equal to "", the first return will be true if the consumer has timed out and should be dropped,
+    // or false otherwise.
+    pure def TimeoutDueToVSCTimeout(currentState: ProtocolState, consumer: Chain): (bool, str) =
+        // check for errors: the consumer is not running
+        if (not(isCurrentlyConsumer(consumer, currentState.providerState))) {
+            (false, "Consumer is not currently a consumer - must have 'running' status!")
+        } else {
+            val providerState = currentState.providerState
+            val consumerState = currentState.consumerStates.get(consumer)
+
+            // has a packet been sent on the provider more than VSCTimeout ago, but we have not received an answer since then?
+            val sentVSCPackets = providerState.sentVSCPackets.get(consumer)
+            if(sentVSCPackets.length() > 0) {
+                val oldestSentVSCPacket = sentVSCPackets.head() // if length is 0, this is undefined, but we check for this before we use it
+                if(oldestSentVSCPacket.sendingTime + VscTimeout < providerState.chainState.lastTimestamp) {
+                    (true, "")
+                } else {
+                    // no timeout yet, it has not been VscTimeout since that packet was sent
+                    (false, "")
+                }
+            } else {
+                // no packet has been sent yet, so no timeout
+                (false, "")
+            }
+        }
+
+    // ===================
+    // ASSUMPTIONS ON MODEL PARAMETERS
```

Comment:
Is there a way to force those tests to run when somebody `import`s `CCV` and binds values to the corresponding constant? If not, maybe it would make sense to do the testing by extending `run_invariants.sh`.
My concern is that if those are not tested, then someone could update `ccv_statemachine.qnt` with wrong values and hence those tests do not add much value.

Reply:
I will add it to `run_invariants.sh`
What you asked is not possible right now, but the Quint team has it on the radar :) https://github.com/informalsystems/quint/issues/1182